src/System.Net.Security/src/System/Net/SecureChannel.cs

```diff
@@ -646,7 +646,10 @@ private bool AcquireClientCredentials(ref byte[] thumbPrint)
                 // We can probably do some optimization here. If the selectedCert is returned by the delegate
                 // we can always go ahead and use the certificate to create our credential
                 // (instead of going anonymous as we do here).
-                if (sessionRestartAttempt && cachedCredentialHandle == null && selectedCert != null)
+                if (sessionRestartAttempt &&
```

Comment:
Nit: I'm having trouble understanding the comment above this.  Is the comment about optimization referring to an optimization we're already doing here, or is it referring to some potential future TODO?  I'm assuming it's the former and "we can probably do some optimization here" really means "under certain circumstances we can take an optimized path."... is that correct?